SCons/Scanner/Python.py

```diff
@@ -82,6 +82,40 @@ def find_include_names(node):
     return all_matches
 
 
+def find_import(import_path, search_paths):
+    """
+    Finds the specified import in the various search paths.
+    For an import of "p", it could either result in a file named p.py or
+    p/__init__.py. We can't do two consecutive searches for p then p.py
+    because the first search could return a result that is lower in the
+    search_paths precedence order. As a result, it is safest to iterate over
+    search_paths and check whether p or p.py exists in each path. This allows
+    us to cleanly respect the precedence order.
+
+    If the import is found, returns a tuple containing:
+        1. Discovered dependency node (e.g. p/__init__.py or p.py)
+        2. True if the import was a package, False if the import was a module.
+        3. The Dir node in search_paths that the import is relative to.
+    If the import is not found, returns a tuple containing (None, False, None).
+    Callers should check for failure by checking whether the first entry in the
+    tuple is not None.
+    """
+    for search_path in search_paths:
+        paths = [search_path]
+        # Note: if the same import is present as a package and a module, Python
+        # prefers the package. As a result, we always look for x/__init__.py
+        # before looking for x.py.
+        node = SCons.Node.FS.find_file(import_path + '/__init__.py', paths)
```

Comment:
Looking through this PR since it's functionality we'd like!  Can Windows handle `/`?  The usual python, OS independent, way of handling paths is `os.path.join`?

Reply:
Good question. I did validate that the scanner and tests work on Windows, so yes I believe it's okay. As far as I can tell, `SCons.Node.FS.find_file` supports `/` because that could come into play when other scanners are used. My understanding is that the C scanner reading a line like `#include <foo/hello.h>` would trigger a similar type of filename being passed to `find_file`.

That said, either way is fine. I just tried replacing all usage of `/` with `os.sep` and all tests succeeded. So I'm open to whatever people prefer.

Reply:
os.path.join() is probably "correct", but yes, both windows, python, and SCons can handle "/"

Reply:
I'd want to avoid using os.path.join() because it has extra logic for absolute paths unrelated to the goal here. For example, the docs say:
```
If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.
```
I am fine replacing `/` with `os.sep` though. Just let me know if you all want me to do that.